test/system/executable/tests.py

```diff
@@ -229,4 +259,17 @@ def test_ddsrouter_closure(ddsrouter, configuration_file):
             'executable permissions.')
         sys.exit(1)
 
-    sys.exit(test_ddsrouter_closure(args.exe, args.config_file))
+    if args.signal is None:
+        logger.error(
+            'Must specify signal to kill process.')
+        sys.exit(1)
+
+    use_sigterm = False
+    if "sigterm" in args.signal:
+        use_sigterm = True
```

Comment:
Move this logic to the `check_terminate_signal()` function and don't forget to check for uppercase inputs also.